changelog.d/20250715_141115_max.tuecke_sc_41884_replace_scope_collection_type.rst

```diff
@@ -0,0 +1,5 @@
+Changed
+-------
+
+- Replaced the ``globus_sdk._types.ScopeCollectionType`` alias with explicit ``str | Scope | t.Iterable[str | Scope]`` type annotations. (:pr:`1259`)
+    - Removed support for nested scope collections in the scope normalization functions (e.g., nested iterables like ``[["scope1"], "scope2"]``).
```

Comment:
Since users should never need to know about an internal detail like `globus_sdk._types`, I'd rather document this in a user-facing way. I'd favor just documenting the user-facing change. Something like,

> Remove support for normalizing nested iterables of scopes, e.g. `[["scope1"], "scope2"]`